### DISCUSSION TITLE
Migration feature requests

### DISCUSSION THREAD

**Discussion by bgiromini** [ServiceStack/Discuss#72] [Ideas]
I think the migrations feature is very nice, but have a few request.

1.  Would like a way to test the migration before running it.  Something like --AppTasks=migrate:test
2.  Consider providing a UI for the migration, maybe a simple screen that lists what migrations to be done, buttons to migrate, revert, test, etc.

**Comment by mythz** [comment]
Not sure what "testing" the migrations would do, if you want to make changes to the latest migration you're working on you would [revert the last and rerun the migration](https://docs.servicestack.net/ormlite/db-migrations#revert-and-rerun-last-migration), which you can do from the included npm scripts or NUnit tests. Otherwise you can already [Rerun migrations individually](https://docs.servicestack.net/ormlite/db-migrations#running-migrations-individually) which are run without any impact on the state in the migration table.

We're not going to provide a UI as migration tasks should be run when the App isn't running, best we can do is provide additional tests to perform any actions or inspect the migration table which you can already view from the [Admin Database UI](https://docs.servicestack.net/admin-ui-database). If you prefer you can add additional tests to the [Migration Tasks](https://docs.servicestack.net/ormlite/db-migrations#run-or-debug-migrations-from-your-ide) to query the `Migration` table directly, e.g you can view the migrations run with:

```csharp
[Test]
public void View_Migrations_Run()
{
    using var db = ResolveDbFactory().Open();
    db.Select<Migration>().OrderBy(x => x.Id).Map(x => x.Name).PrintDump();
}
```
